src/cpp/thread_manager/thread_manager.cc

```diff
@@ -166,22 +166,37 @@ void ThreadManager::MainWorkLoop() {
       case WORK_FOUND:
         // If we got work and there are now insufficient pollers and there is
         // quota available to create a new thread, start a new poller thread
-        if (!shutdown_ && num_pollers_ < min_pollers_ &&
-            grpc_resource_user_allocate_threads(resource_user_, 1)) {
-          num_pollers_++;
-          num_threads_++;
-          if (num_threads_ > max_active_threads_sofar_) {
-            max_active_threads_sofar_ = num_threads_;
+        bool got_thread;
```

Comment:
init to true then only override in the false case below?

Reply:
I thought about that, but leaving it uninitialized at first makes sure that we're covering all our cases properly (since the compiler would complain otherwise about use without initialization). I didn't want to accidentally leave it in a bad state. But I can also see that it's probably more idiomatic to do what you said. That said, I will instead reverse the condition (to be `resource_exhausted` and start it as false).